File: ECommerce-Bill_Feature/src/app/models/OrderItem.ts
import { Offer } from "./Offer";
import { Order } from "./Order";
import { Product } from "./Product";

export class OrderItem {
    id: number;
    orderId: number | null;
    productId: number | null;
    productName: string;
    price: number | null;
    quantity: number | null;
    subTotal: number | null;
    offerId: number | null;
    isReturned: boolean | null;
    returnedOn: string | null;
    offer: Offer;
    order: Order;
    product: Product;
}

File: ECommerceAPI/Models/OrderItem.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace ECommerceAPI.Models
{
    public partial class OrderItem
    {
        public int Id { get; set; }
        public long? OrderId { get; set; }
        public string ProductName { get; set; }
        public int? ProductId { get; set; }
        public double? Price { get; set; }
        public int? Quantity { get; set; }
        public double? SubTotal { get; set; }
        public int? OfferId { get; set; }
        public bool? IsReturned { get; set; }
        public DateTime? ReturnedOn { get; set; }

        public virtual Offer Offer { get; set; }
        public virtual Order Order { get; set; }
        public virtual Product Product { get; set; }
    }
}
